tests/robot-tests/tests/admin_and_public/bau/subject_reordering.robot

```diff
@@ -0,0 +1,246 @@
+    user checks accordion is in position    Two    2    id:uploadedDataFiles
+    user checks accordion is in position    Three    3    id:uploadedDataFiles
+    user checks accordion is in position    Four    4    id:uploadedDataFiles
+
+Complete data replacement
+    user opens accordion section    Three
+    ${THREE_CONTENT}    user gets accordion section content element    Three    id:uploadedDataFiles
+    user clicks link    Replace data    ${THREE_CONTENT}
+
+    user waits until page contains    Data blocks: OK
+    user waits until page contains    Footnotes: OK
+    user clicks button    Confirm data replacement
+    user waits until h2 is visible    Data replacement complete
+
+Validate subject order is correct after replacement
+    user clicks link    Data and files
+    user waits until page contains element    id:uploadedDataFiles
+
+    user checks accordion is in position    One    1    id:uploadedDataFiles
+    user checks accordion is in position    Two    2    id:uploadedDataFiles
+    user checks accordion is in position    Three    3    id:uploadedDataFiles
+    user checks accordion is in position    Four    4    id:uploadedDataFiles
+
+Add data guidance for all subjects
+    user clicks link    Data guidance
+    user waits until h2 is visible    Public data guidance
+
+    user waits until page contains element    id:dataGuidance-dataFiles
+    user waits until page contains accordion section    One
+
+    user waits until page contains element    id:dataGuidance-dataFiles
+    user checks accordion is in position    One    1    id:dataGuidance-dataFiles
+    user checks accordion is in position    Two    2    id:dataGuidance-dataFiles
+    user checks accordion is in position    Three    3    id:dataGuidance-dataFiles
+    user checks accordion is in position    Four    4    id:dataGuidance-dataFiles
+
+    user enters text into data guidance data file content editor    One
+    ...    One guidance content
+    user enters text into data guidance data file content editor    Two
+    ...    Two guidance content
+    user enters text into data guidance data file content editor    Three
+    ...    Three guidance content
+    user enters text into data guidance data file content editor    Four
+    ...    Four guidance content
+
+    user clicks button    Save guidance
+
+Publish release
+    user approves original release for immediate publication
+
+Check subject order in data tables
+    user navigates to data tables page on public frontend
+
+    user clicks radio    %{TEST_THEME_NAME}
+    user clicks radio    ${PUBLICATION_NAME}
+    user clicks element    id:publicationForm-submit
+
+    user waits until table tool wizard step is available    2    Choose a subject
+    user checks previous table tool step contains    1    Publication    ${PUBLICATION_NAME}
+
+    user checks page contains element
+    ...    xpath://*[contains(@class, "govuk-radios__item")][1]//label[contains(text(), "One")]
```

Comment:
Would extract a keyword like `user checks radio exists`. 

Would also use a better selector more inline with other radio keywords `xpath://label[text()="${label}"]/../input[@type="radio"]` where we don't rely on a class selector like `govuk-radios__item`

Reply:
I've extracted this to the new keyword `user checks x position radio has label`.

In order to avoid the `govuk-radios__item`, i've had to resort to `*[contains(@data-testid, "Radio item for ")]`. This is necessary as I need to select the Xth radio first, then check the label. I tried starting the selector an `input`, but `//input[@type="radio"][2]` didn't work.